specification/hdinsight/data-plane/Microsoft.HDInsight/preview/2018-11-01-preview/livySpark.json

```diff
@@ -0,0 +1,1128 @@
+          }
+        },
+        "queue": {
+          "type": "string"
+        },
+        "name": {
+          "type": "string"
+        },
+        "conf": {
+          "x-ms-client-name": "configuration",
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          }
+        },
+        "heartbeatTimeoutInSecond": {
+          "format": "int32",
+          "type": "integer"
+        }
+      }
+    },
+    "SparkJobLog": {
+      "type": "object",
+      "properties": {
+        "id": {
+          "format": "int32",
+          "type": "integer"
+        },
+        "from": {
+          "format": "int32",
+          "type": "integer"
+        },
+        "size": {
+          "format": "int32",
+          "type": "integer"
+        },
+        "log": {
+          "x-ms-client-name": "logLines",
+          "uniqueItems": false,
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        }
+      }
+    },
+    "SparkJobState": {
+      "type": "object",
+      "properties": {
+        "id": {
+          "format": "int32",
+          "type": "integer"
+        },
+        "state": {
+          "type": "string"
+        }
+      }
+    },
+    "SparkStatementCollection": {
+      "type": "object",
+      "properties": {
+        "statements": {
+          "uniqueItems": false,
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/SparkStatement"
+          }
+        }
+      }
+    },
+    "SparkStatement": {
+      "type": "object",
+      "properties": {
+        "id": {
+          "format": "int32",
+          "type": "integer"
+        },
+        "code": {
+          "type": "string"
+        },
+        "state": {
+          "type": "string"
+        },
+        "output": {
+          "$ref": "#/definitions/SparkStatementOutput"
+        }
+      }
+    },
+    "SparkStatementOutput": {
+      "type": "object",
+      "properties": {
+        "status": {
+          "type": "string"
+        },
+        "execution_count": {
+          "format": "int32",
+          "type": "integer"
+        },
+        "data": {
+          "type": "object"
+        }
+      }
+    },
+    "SparkStatementRequest": {
+      "type": "object",
+      "properties": {
+        "code": {
+          "type": "string"
+        },
+        "kind": {
+          "type": "string"
+        }
+      }
+    },
+    "SparkStatementCancellationResult": {
+      "type": "object",
+      "properties": {
+        "msg": {
+          "x-ms-client-name": "cancelMessage",
+          "type": "string"
+        }
+      }
+    }
+  },
+  "parameters": {
+    "requestedByParameter": {
+      "name": "X-Requested-By",
+      "x-ms-parameter-location": "client",
+      "x-ms-client-name": "requestedBy",
+      "description": "Add default value for X-Requested-By in header.",
+      "in": "header",
+      "required": false,
+      "default": "ambari",
```

Comment:
Sounds a strange default value?

Reply:
Set  "X-Requested-By" as "ambari"  is a workaround because there is a check in Livy service. And this value has been verified by other project. [Project Link](https://github.com/microsoft/azure-tools-for-java/blob/0b61e5dee7dd108611e28c6617d36d120f192d70/Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/spark/common/SparkBatchSubmission.java#L250)